src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2807,68 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
 #endif
 }
 
+TEST_CASE("LedgerKeyMeter tests")
```

Comment:
These tests are off to a good start and are much easier to follow! I think we should add one more end-to-end test case in this file, you probably want to test the following sections inside of it:
1. Prefetch a mix of soroban and classic keys, all keys have enough quota
2. Prefetch a mix of keys, some keys don't have quota
3. Prefetch a mix of keys, keys have quota, but some indiviual TXs don't have quota.

Ideally, these tests would be as high level as possible.